src/main/scala/com/cognite/spark/datasource/EventsRelation.scala

```diff
@@ -136,13 +139,111 @@ class EventsRelation(config: RelationConfig)(@transient val sqlContext: SQLConte
     (postUpdate, postNewItems).parMapN((_, _) => ())
   }
 
+  sealed case class TypeFilter(`type`: String)
+
+  // scalastyle:off cyclomatic.complexity
+  def getTypeFilters(filter: Filter): Seq[TypeFilter] =
```

Comment:
Is it possible to combine the two by making it a function of `(filter: Filter, columnName: String, mkFilter: (String) => ColumnFilter)` (not tested) or similar? And make `TypeFilter` and `SubtypeFilter` subclasses of `ColumnFilter`.